.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,326 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+
+env:
+  BUNDLE_CLEAN: true
+  BUNDLE_FROZEN: true
+  BUNDLE_WITHOUT: 'development production'
+  BUNDLE_DEPLOYMENT: true
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    env:
+      RAILS_ENV: test
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          cache: yarn
+          node-version-file: '.nvmrc'
+
+      - name: Install native Ruby dependencies
+        run: sudo apt-get install -y libicu-dev libidn11-dev
+
+      - name: Set up bundler cache
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: .ruby-version
+          bundler-cache: true
+
+      - run: yarn install --frozen-lockfile
+      - name: Precompile assets
+        # Previously had set this, but it's not supported
+        # export NODE_OPTIONS=--openssl-legacy-provider
```

Comment:
Not sure what this one was used for, as current Node throws a warning/error using this flag in the variable

Reply:
This was used because OpenSSL 3 doesn't support some of the algorithms used by webpack. Without it, on some node versions, this would result in the build failing with loads of errors like this:
```
ERROR in chunk public [initial]
js/public-73f580d0c8328bbd755f.chunk.js
/home/claire/Sources/GIT/mastodon/node_modules/babel-loader/lib/index.js??ref--8-0!/home/claire/Sources/GIT/mastodon/app/javascript/mastodon/load_keyboard_extensions.js
error:0308010C:digital envelope routines::unsupported
```

Reply:
Thanks for pointing that out. Looks like it's not an issue/supported, because the version in `.nvmrc` is still on the v16 release line, and the OpenSSL 3.0 was a 17+ thing https://stackoverflow.com/questions/69962209/what-is-openssl-legacy-provider-in-node-js-v17